formant_ros2_adapter/scripts/main.py

```diff
@@ -348,42 +353,84 @@ def setup_ros2_params(self):
         print("INFO: Setting new ROS2 parameters")
         self.ros2_node.set_parameters(new_params)
 
-    def setup_subscribers(self):
+    def get_active_topics(self):
+
+        topic_names_and_types = self.ros2_node.get_topic_names_and_types()
+
+        active_topics = []
+        for topic_name, _ in topic_names_and_types:
+            publishers_info = self.ros2_node.get_publishers_info_by_topic(topic_name)
+            if publishers_info:
+                active_topics.append(topic_name)
+
+        return active_topics
+
+    def _setup_subscribers(self):
         # Remove any existing subscribers
         for subscriber_list in self.ros2_subscribers.values():
             for subscriber in subscriber_list:
                 self.ros2_node.destroy_subscription(subscriber)
 
         self.ros2_subscribers = {}
+        self.topics_to_subscribe = [
+            _.get("ros2_topic") for _ in self.config.get("subscribers", [])
+        ]
+        print("Topics from config:")
+        print(self.topics_to_subscribe)
+        self._schedule_subscribe_to_topics()
+
+    def _schedule_subscribe_to_topics(self):
+        t = threading.Timer(1.0, self.setup_subscribers_inner)
+        t.daemon = True
+        t.start()
+
+    def setup_subscribers_inner(self):
+        with self._subscribe_mutex:
+            """Get all active topics in ROS 2 runtime"""
+            print("Topics in runtime")
+            topics_in_runtime = self.get_active_topics()
+            print(topics_in_runtime)
+
+            for subscriber_config in self.config.get("subscribers", []):
+                subscriber_topic = subscriber_config["ros2_topic"]
+                """For each subscriber, check if it is in the list of topics that don't already have a subscriber created"""
+                if subscriber_topic in self.topics_to_subscribe:
+                    """If the topic is in the active topics, create a subscriber"""
+                    if subscriber_topic in topics_in_runtime:
+                        print(f"Subscriber for topic: {subscriber_topic}")
+                        self.topics_to_subscribe.remove(subscriber_topic)
+
+                        if "ros2_qos_profile" in subscriber_config:
+                            subscriber_qos = QOS_PROFILES.get(
+                                subscriber_config["ros2_qos_profile"],
+                                qos_profile_system_default,
+                            )
+                        else:
+                            subscriber_qos = qos_profile_system_default
+                        print(f"QoS: {subscriber_qos}, {type(subscriber_qos)}")
```

Comment:
I think we can reduce or remove this statement, that's left over from way long ago when I was debugging